### ISSUE TITLE
postgresql_database idempotency

### ISSUE THREAD

**Issue by francocatena** [sous-chefs/postgresql#533]
### Cookbook version
7.0.0

### Chef-client version
14.1.12

### Platform Details
Ubuntu 18.04

### Scenario:
Create a database

### Steps to Reproduce:
```ruby
postgresql_user 'test_user' do
  password  '123'
end

postgresql_database 'test_db' do
  owner 'test_user'
end
````

### Expected Result:
Database creation on first run, no action on subsequent runs.

### Actual Result:
It correctly creates the DB on the first run, but later fails to detect it:

```
STDERR: createdb: database creation failed: ERROR: database "database" already exists
```

It seems the same error as #514, creating a new one since version has changed and the other one is considered closed.

**Comment by jldugger** [comment]
I'm running into the same issue in test kitchen with 16.04.2 LTS.

**Comment by damacus** [comment]
Thanks for this report. I've fixed this occurrence locally, but it's surfaced a problem with a few other commands. Apologies the fix could take a little longer!

**Comment by florian-asche** [comment]
Sry. Did not see that problem at my tests. But occurred now. I added a file check as temp solution...

**Comment by DWSR** [comment]
Would it not make sense to do something like this to check if the database exists? https://stackoverflow.com/questions/18389124/simulate-create-database-if-not-exists-for-postgresql

**Comment by damacus** [comment]
@DWSR that works better outside the world of config management. Right now we want to  do the `check_if_exists` outside the block creating the database a la: 

https://github.com/sous-chefs/postgresql/blob/ISSUE-533/libraries/helpers.rb#L58

This now has unit tests and uses the "standard" connect syntax, so I'm pretty happy with it.

**Comment by damacus** [comment]
Closed as part of #535

**Comment by damacus** [comment]
This will be released as part of 7.1
